pkg/reboot/calculator.go

```diff
@@ -23,102 +23,106 @@ const (
+	GetRebootDuration(k8sClient client.Client, ctx context.Context, node *v1.Node) (time.Duration, error)
+	// SetConfig sets the SelfNodeRemediationConfig to be used for calculating the minimum reboot duration
+	SetConfig(config *v1alpha1.SelfNodeRemediationConfig)
 }
 
-type safeTimeCalculator struct {
-	timeToAssumeNodeRebooted, minTimeToAssumeNodeRebooted                   time.Duration
-	wd                                                                      watchdog.Watchdog
-	maxErrorThreshold                                                       int
-	apiCheckInterval, apiServerTimeout, peerDialTimeout, peerRequestTimeout time.Duration
-	log                                                                     logr.Logger
-	k8sClient                                                               client.Client
-	highestCalculatedBatchNumber                                            int
-	isAgent                                                                 bool
+var _ Calculator = &calculator{}
+
+type calculator struct {
+	k8sClient client.Client
+	log       logr.Logger
+	// storing the config here when reconciling it increases resilience in case of issues during remediation
+	snrConfig *v1alpha1.SelfNodeRemediationConfig
 }
 
-func NewAgentSafeTimeCalculator(k8sClient client.Client, wd watchdog.Watchdog, maxErrorThreshold int, apiCheckInterval, apiServerTimeout, peerDialTimeout, peerRequestTimeout, timeToAssumeNodeRebooted time.Duration) SafeTimeCalculator {
-	return &safeTimeCalculator{
-		wd:                       wd,
-		maxErrorThreshold:        maxErrorThreshold,
-		apiCheckInterval:         apiCheckInterval,
-		apiServerTimeout:         apiServerTimeout,
-		peerDialTimeout:          peerDialTimeout,
-		peerRequestTimeout:       peerRequestTimeout,
-		timeToAssumeNodeRebooted: timeToAssumeNodeRebooted,
-		k8sClient:                k8sClient,
-		isAgent:                  true,
-		log:                      ctrl.Log.WithName("safe-time-calculator"),
+func NewCalculator(k8sClient client.Client, log logr.Logger) Calculator {
+	return &calculator{
+		k8sClient: k8sClient,
+		log:       log,
 	}
 }
 
-func NewManagerSafeTimeCalculator(k8sClient client.Client, timeToAssumeNodeRebooted time.Duration) SafeTimeCalculator {
-	return &safeTimeCalculator{
-		timeToAssumeNodeRebooted: timeToAssumeNodeRebooted,
-		k8sClient:                k8sClient,
-		isAgent:                  false,
-		log:                      ctrl.Log.WithName("safe-time-calculator"),
-	}
+func (r *calculator) SetConfig(config *v1alpha1.SelfNodeRemediationConfig) {
+	r.snrConfig = config
 }
 
-func (s *safeTimeCalculator) GetTimeToAssumeNodeRebooted() time.Duration {
-	if !s.isAgent {
-		return s.timeToAssumeNodeRebooted
+// TODO add unit test!
+func (r *calculator) GetRebootDuration(k8sClient client.Client, ctx context.Context, node *v1.Node) (time.Duration, error) {
```

Comment:
please remove k8sClient from method params

Reply:
hm, Goland's unused var / function warning used to work better... thx, fixed

Reply:
NP.
in this case I don't think it was considered an unused var because the method used the variable passed to it and not the one encapsulated in the struct.